mindmeld/models/model_factory.py

```diff
@@ -0,0 +1,186 @@
+
+        Returns:
+            model (Type[AbstractModel]): A text/tagger model instance
+
+        Raises:
+            ValueError: When the configs are invalid
+        """
+
+        if not (model_config and isinstance(model_config, (ModelConfig, dict))):
+            msg = f"Need a valid model config to create a text/tagger model in ModelFactory. " \
+                  f"Found model_config={model_config} of type({type(model_config)})"
+            raise ValueError(msg)
+
+        # get model type upon validation
+        model_config = cls._resolve_model_config(model_config)
+        model_type = cls._get_model_type(model_config)
+
+        # instantiate and return appropriate model class from the model type
+        model_class = ModelFactory.MODEL_TYPE_TO_FACTORY[model_type].get_model_class(model_config)
+        return model_class(model_config)
+
+    @classmethod
+    def create_model_from_path(cls, path: str) -> Union[None, Type[AbstractModel]]:
+        """
+        Loads and returns a model from the specified path
+
+        Args:
+            path (str): A pickle file path from where a model can be loaded
+
+        Returns:
+            model (Union[None, Type[AbstractModel]]): Returns None when the specified path is not
+                found or if the model loaded from the specified path is a NoneType. If found a valid
+                config and a valid model, the model is load by calling .load() method and returned
+
+        Raises:
+            ValueError: When the path is invalid
+        """
+
+        if not (path and isinstance(path, str)):
+            msg = f"Need a valid path to load a text/tagger model in ModelFactory. " \
+                  f"Found path={path} of type({type(path)})"
+            raise ValueError(msg)
+
+        if not path.endswith(".pkl"):
+            msg = "Model Path must end with .pkl for ModelFactory to be able to identify the model"
+            raise ValueError(msg)
+
+        try:
+            # if loading from path, determine the ABCModel type & return after doing xxxModel.load()
+            model_config = AbstractModel.load_model_config(path)
+
+            # get model type upon validation
+            model_config = cls._resolve_model_config(model_config)
+            model_type = cls._get_model_type(model_config)
+
+            # load metadata and return
+            model_class = ModelFactory.MODEL_TYPE_TO_FACTORY[model_type].get_model_class(
+                model_config)
+            return model_class.load(path)
+
+        except FileNotFoundError:
+            # sometimes a model (and its config file) might not be dumped, eg. in role classifiers
+            # or even if dumped, can be of NoneType enclosed in a dictionary
+            msg = f"No model file found while trying to load model from path: {path}. It might " \
+                  f"be the case that the classifier didn't need a model due to one or no classes."
+            logger.warning(msg)
+            return None
```

Comment:
Consider removing the `return None`. Also should we throw an error if the file is not found instead of silently returning None?

Reply:
Hey Kunal, so I had to return None and deal with the result (NoneType or not) directly in components/ folder because of backwards compatibilities wrt the way we store and load sklearn models. I specifically set the `return None` because a NoneType output here explicitly means no model has been found and the classifier.py, role_classifier.py and entity_recognizer.py utilizes this info. 

The reason I did not raise the error and catch it as a means to obtain the same info is because, in mindmeld's architecture, helpers.py's load_model() acts as a bridge between components/ files and models/ files and the two modules don't communicate directly nor have to, hence no dependencies. So in order to propogate the presence of model information from models/ to components/, one design choice i thought is that the components/ files know that a FileNotFoundError is raised by models/ files and do a try-catch for that specific Error type, but then what if we change the Error type in future, we'll not be able to track and debug the dependency created across modules right? Alternate choice i thought is to just return a NoneType, which can remain same irrespective of what error type we create. This will not make the components/ files dependent on the choices of models/ files and will be easy to debug later on. Creating a dummy model class for this purpose seemed to complicate things and I refrained from that idea.

What do you think of this? Please let me know if you think I can do a better solution. And Thanks for pointing this out.